### PR TITLE
EKS Overrides for AWS Batch submit_job

### LINKED ISSUE #40117: EKS Overrides for AWS Batch submit_job
### Body

Similar to https://github.com/apache/airflow/pull/39903 but for EKS

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by ssilb4**
aws batch operator has no eks_properties_override. so I added it.

I found [this](https://github.com/apache/airflow/pull/39903)

Closes: https://github.com/apache/airflow/issues/40117

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Review by o-nikolas — COMMENTED** [review]
Tests are failing from code formatting issues, but otherwise lgtm

**Comment by ssilb4** [comment]
I changed test

File: airflow/providers/amazon/aws/hooks/batch_client.py [hunk 1]
```diff
@@ -103,6 +103,7 @@ def submit_job(
         parameters: dict,
         containerOverrides: dict,
         ecsPropertiesOverride: dict,
+        eksPropertiesOverride: dict,
```
Comment (vincbeck) [inline airflow/providers/amazon/aws/hooks/batch_client.py:106]: Please make it optional to avoid adding breaking change.

**Review by vincbeck — CHANGES_REQUESTED** [review]

File: airflow/providers/amazon/aws/hooks/batch_client.py (hunk 1 above)
Comment (ssilb4) [reply airflow/providers/amazon/aws/hooks/batch_client.py:106]: you mean Optional[dict] ? then, other parameters change same?

Comment (vincbeck) [reply airflow/providers/amazon/aws/hooks/batch_client.py:106]: Actually, from #39903, it seems it is not a breaking change ... You can ignore this comment :) Sorry about that

**Review by vincbeck — APPROVED** [review]

**Comment by ssilb4** [comment]
I think non-db test failure is not my fault.

**Comment by potiuk** [comment]
Yep - it's being fixed in https://github.com/apache/airflow/pull/40821

**Comment by vincbeck** [comment]
> Yep - it's being fixed in #40821

Are you sure? I have the same issue in my PR #40823 and it is failing (fork and PR are rebased)

**Review by eladkal — APPROVED** [review]

**Pull request merged**